api/src/main/java/com/spotify/ffwd/output/OutputPlugin.java

```diff
@@ -62,35 +69,38 @@ public OutputPlugin(
      * <code>output := new FilteringPluginSink(filter, delegator:=output)</code>
      *
      * @param input binding key with injection type of plugin sink
-     * @param output binding key, containing injection type of wrapping plugin sink
+     * @param wrapKey binding key, containing injection type of wrapping plugin sink
      * @return module that exposes output binding key
      */
     protected Module wrapPluginSink(
-        final Key<? extends PluginSink> input, final Key<PluginSink> output
+        final Key<? extends PluginSink> input, final Key<PluginSink> wrapKey
     ) {
         return new PrivateModule() {
             @Override
             protected void configure() {
                 Key<PluginSink> sinkKey = (Key<PluginSink>) input;
 
-                if (flushInterval != null && flushInterval.isPresent() &&
-                    BatchedPluginSink.class.isAssignableFrom(
-                        sinkKey.getTypeLiteral().getRawType())) {
+                if (batching.getFlushInterval().isPresent() &&
+                    BatchablePluginSink.class.isAssignableFrom(
+                        input.getTypeLiteral().getRawType())) {
                     final Key<PluginSink> flushingKey =
                         Key.get(PluginSink.class, Names.named("flushing"));
-                    final Key<? extends BatchedPluginSink> batchedPluginSink =
-                        (Key<? extends BatchedPluginSink>) (Key<? extends PluginSink>) sinkKey;
+                    // IDEA doesn't like this cast, but it's correct, tho admittedly not pretty
+                    final Key<? extends BatchablePluginSink> batchedPluginSink =
+                        (Key<? extends BatchablePluginSink>) (Key<? extends PluginSink>) sinkKey;
 
                     // Use annotation so that we can avoid name space clash
-                    bind(BatchedPluginSink.class)
-                        .annotatedWith(FlushingDelegate.class)
+                    bind(BatchablePluginSink.class)
+                        .annotatedWith(BatchingDelegate.class)
                         .to(batchedPluginSink);
-                    bind(flushingKey).toInstance(new FlushingPluginSink(flushInterval.get()));
+                    bind(flushingKey).toInstance(
+                        new BatchingPluginSink(batching.getFlushInterval().get(),
+                            batching.getBatchSizeLimit(), batching.getMaxPendingFlushes()));
 
                     sinkKey = flushingKey;
                 }
 
-                if (filter != null && filter.isPresent()) {
```

Comment:
filter can be empty

Reply:
I don't think so. It's an optional, so it will be Optional.empty() if it wasn't specified in the configuration. It will not be null.